R/editing.R

```diff
@@ -356,6 +356,16 @@ publish_update <- function(mn,
       stopifnot(object_exists(mn, parent_child_pids))
   }
 
+  # Check that resource_map_pid, and metadata_pid are the current versions
+  meta_obsoletedBy <- dataone::getSystemMetadata(mn, metadata_pid)@obsoletedBy
```

Comment:
Other checks in this function are guarded by the value fo the `check_first` arg. Is this one and, if not, could it be?

These two calls alone could slow down an `publish_update` by 2-10 seconds depending on server load.

Reply:
`check_first` just checks if the objects exist, not if they're already obsoleted.  It certainly could be put into the `check_first` block though.  

I would argue against that (at least for the resource map check) because you can set to `check_first = FALSE` and using an obsoleted pid by mistake is one of our most common errors.  It's harmless to use an incorrect metadata pid, but using an incorrect resource map pid will actually obsolete the metadata first before it fails, leaving the package as stranded metadata.  This isn't a problem for the experienced user to fix, but new users typically make matters worse before asking for help, and we spend lots of time figuring out what happened.  

In short:  moving the metadata pid check into the `check_first` block is ok, but an incorrect resource map pid causes problems that I'd rather avoid.  What do you think?  


MRE: 
```
## Incorrect metadata pid 
pkg1 <- create_dummy_package(adc_test)
pkg2 <- publish_update(adc_test, pkg1$metadata, pkg1$resource_map, pkg1$data)

# update package with incorrect metadata pid: 
pkg3 <- publish_update(adc_test, pkg1$metadata, pkg2$resource_map, pkg2$data) # this call errors but is harmless
# We can still publish successfully 
pkg3 <- publish_update(adc_test, pkg2$metadata, pkg2$resource_map, pkg2$data)


## Incorrect resource map 
pkg1 <- create_dummy_package(adc_test)
pkg2 <- publish_update(adc_test, pkg1$metadata, pkg1$resource_map, pkg1$data)

# update package with incorrect resource map pid:  
pkg3 <- publish_update(adc_test, pkg2$metadata, pkg1$resource_map, pkg2$data) 
# this call errors but obsoletes the metadata beforehand 
browseURL(paste0("https://test.arcticdata.io/#view/", pkg2$metadata))
# we also can't publish correctly now: 
pkg3 <- publish_update(adc_test, pkg2$metadata, pkg2$resource_map, pkg2$data)
```

Reply:
Excellent points! I think I could go either way on this. Always doing these two checks each time the function is run is fine w/ me if you prefer that.